File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.js
function solution(elements) {
    var answer = 0;
    
    let set = new Set();
    const len = elements.length;
    for(let i=0; i<len; i++) {
            let sum =0;
            
            for(let j=i; j<i+len; j++) {
                sum += elements[j%len];
                set.add(sum);
            }
        }
    
    answer = set.size;
    
    return answer;
}

File: 프로그래머스/2/131701. 연속 부분 수열 합의 개수/연속 부분 수열 합의 개수.cpp
#include <string>
#include <vector>
#include <set>

using namespace std;

int solution(vector<int> elements) {
    int answer = 0;
    
    int len = elements.size();
    
    for(int i=0; i<len ;i++) 
    {
        elements.push_back(elements[i]);
    }
    
    set<int> nums;
    
    for(int num=1; num<=len; num++) 
    {
        
        for(int i=0; i<len; i++) 
        {
            int total =0;
            for(int j=i; j<i+num; j++) 
            {
                total += elements[j];
            }
            
            nums.insert(total);
        }
    }
    
    answer = nums.size();
    
    return answer;
}